### ISSUE TITLE
Standardize encoding/decoding of byte data on Java and Python sides

### ISSUE THREAD

**Issue by henrypinkard** [micro-manager/pycro-manager#200]
Right now they use slightly different mechanisms. Should data type be encoded in the key of the JSON message itself, or at a higher level?

**Comment by binli123** [comment]
I ran into a bug that might be associated with this. I got an error when trying to use image_process_fn in Acquisition which says:
``` 
File "C:\Users\lociuser\Miniconda3\envs\pyimagej\lib\site-packages\pycromanager\core.py", line 134, in _replace_bytes
    dict_or_list[key].split("@")[1], 16
ValueError: invalid literal for int() with base 16: 'PXI2Slot2'
```
The name is associated with a customized device adaptor for controlling laser scanning devices. Printing the value of dict_or_list[key] at the place where this error occurs gives: OpenScan-NIDAQ@PXI2Slot2, which is the name of the device @ the name of the acquisition card on this device. I think Mark @marktsuchida will know better what this string stands for.

This error does not occur and image_process_fn functions properly when this customized device adaptor is removed and leaving only the camera.

**Comment by henrypinkard** [comment]
I think this is because the encoding makes use of `@` for translating Java-Python. As a workaround, can you change the device adapter name in the mean time. I was hoping to build out a bunch of testing before actually changing any of this, so that I don't break other stuff in the process

**Comment by binli123** [comment]
> I think this is because the encoding makes use of `@` for translating Java-Python. As a workaround, can you change the device adapter name in the mean time. I was hoping to build out a bunch of testing before actually changing any of this, so that I don't break other stuff in the process

Do you mean that the encoding makes use of `@` as a start point for the translation? Does this mean that `@` should be avoided in the device name at this point?

**Comment by henrypinkard** [comment]
Yeah something like that. I honestly don't remember all the details as I made it pretty hastily without advance planning (hence this issue). But Im pretty sure if you take the `@` out of the name it will work.

**Comment by binli123** [comment]
> Yeah something like that. I honestly don't remember all the details as I made it pretty hastily without advance planning (hence this issue). But Im pretty sure if you take the `@` out of the name it will work.

It works! Thanks.
